### PR TITLE
Roll all leadership-expanded components int one shared component

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes ilios/ilios#5347

### CODE REVIEW

File: packages/ilios-common/addon/components/leadership-expanded.js [hunk 1]
```diff
@@ -18,31 +18,61 @@ export default class CourseLeadershipExpandedComponent extends Component {
   }
 
   @cached
-  get courseDirectors() {
-    return new TrackedAsyncData(this.args.course.directors);
+  get modelHasDirectors() {
+    return 'directors' in this.args.model;
```
Comment (jrjohnson) [inline packages/ilios-common/addon/components/leadership-expanded.js:22]: This is terrifying to me and I'm not sure why. Nice though, looks good.

File: packages/ilios-common/addon/components/leadership-expanded.js [hunk 2]
```diff
@@ -18,31 +18,61 @@ export default class CourseLeadershipExpandedComponent extends Component {
...
+    }
+    return null;
   }
 
   get directors() {
-    const directors = this.courseDirectors.isResolved ? this.courseDirectors.value.slice() : [];
+    if (!this.modelHasDirectors) {
+      return [];
+    }
+    const directors = this.modelDirectors.isResolved ? this.modelDirectors.value.slice() : [];
```
Comment (jrjohnson) [inline packages/ilios-common/addon/components/leadership-expanded.js:63]: I'm not sure these need to get sliced. IIRC in Ember data these are arrays now or at least array proxies and don't need special handling anymore.

File: packages/ilios-common/addon/components/leadership-expanded.js (hunk 1 above)
Comment (stopfstedt) [reply packages/ilios-common/addon/components/leadership-expanded.js:22]: yea, it's low-level and a somewhat obscure, feels a bit odd to use. 

i checked the ember-data docs for something that could do introspection, like a 'hasAttribute()' method or the likes, but couldn't find anything.

File: packages/ilios-common/addon/components/leadership-expanded.js (hunk 2 above)
Comment (stopfstedt) [reply packages/ilios-common/addon/components/leadership-expanded.js]: nice catch. removed. we should do that everywhere then.

Comment (stopfstedt) [reply packages/ilios-common/addon/components/leadership-expanded.js]: :heavy_check_mark:

File: packages/ilios-common/addon/components/leadership-expanded.js (hunk 1 above)
Comment (stopfstedt) [reply packages/ilios-common/addon/components/leadership-expanded.js:22]: :shrug:

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**